File: Java/pattern/CommandProj/src/com/pattern/ConcreteCommand.java
package com.pattern;

import javax.sound.midi.Receiver;

public class ConcreteCommand implements ICommand {

	private MyReceiver receiver;
	public ConcreteCommand(MyReceiver receiver) {
		super();
		this.receiver = receiver;
	}
	@Override
	public void exec() {
		receiver.sayHi();
	}

}

File: cplusplus/design_pattern/CommandProj/ConcreteCommand.cpp
#include "stdafx.h"
#include "ConcreteCommand.h"
#include "Receiver.h"


ConcreteCommand::ConcreteCommand()
{
}


ConcreteCommand::~ConcreteCommand()
{
}


ConcreteCommand::ConcreteCommand(std::shared_ptr<Receiver> receiver)
{
}

void ConcreteCommand::exec()
{
    receiver->sayHi();
}
